### DISCUSSION TITLE
Error when installing idh node sensor

### DISCUSSION THREAD

**Discussion by xpowershellx** [Security-Onion-Solutions/securityonion#8154] [Unsupported Versions]
Hello . 
I have a problem when adding a new idh sensor . I have a distributed deployment with 1 manager and 2 sensors which installed/works correctly . The problem is when I try to install an idh sensor . I get the following when "salt-call -l all state.highstate":
Succeeded: 162 
Failed:      2
----------
          ID: so-idh
    Function: docker_image.present
      Result: False
     Comment: Encountered error building /opt/so/saltstack/local/salt/idh as so-idh:latest: Unable to perform build: You must specify a directory to build in path
     Started: 10:06:21.953778
    Duration: 15.059 ms
     Changes:   
----------
          ID: so-idh-run
    Function: docker_container.running
        Name: so-idh
      Result: False
     Comment: Failed to pull so-idh: Error 404: pull access denied for so-idh, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
     Started: 10:06:21.970505
    Duration: 1548.054 ms
     Changes:

On manager:
![image](https://user-images.githubusercontent.com/50677773/174582282-fbf62ce6-75ac-446d-89fc-0a1549bdbde0.png)

On so-status I have in idh node:
Checking Docker status

    Docker -------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    

Checking container statuses

    so-filebeat --------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-sensoroni -------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-telegraf --------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-wazuh ------------------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]

**Comment by defensivedepth** [comment]
Can you confirm your architecture? Doesn't sound like you have a Search Node?

Also, what version are you running on the Manager?

**Reply by xpowershellx** [reply]
Hi.Thanks for the answer .I have a manager search and 2 forward nodes working since over 1 year . Current version  is 2.3.130 .
![Screenshot_20220621-223018_Gallery.jpg](https://user-images.githubusercontent.com/50677773/174882452-c53f8304-6079-475b-86fe-bf1ef41e1afc.jpg)
Let me know if i can provide more info.

**Reply by defensivedepth** [reply]
Run `sudo so-docker-refresh` on the Manager then try the highstate again on the IDH node.

If that still doesn't work, confirm that the idh image is actually on the Manager:

`sudo docker images | grep idh`

**Reply by xpowershellx** [reply]
I did the procedure but it is the same (Run sudo so-docker-refresh on the Manager then try the highstate again on the IDH node.).
And on "sudo docker images | grep idh" i get : 
ghcr.io/security-onion-solutions/so-idh                             2.3.130   21544667efd2   2 weeks ago    227MB
srv-r........:5000/security-onion-solutions/so-idh                  2.3.130   21544667efd2   2 weeks ago    227MB

**Reply by defensivedepth** [reply]
So what is odd is that the `so-status` results from your IDH node don't even show that the IDH container is `missing` or errored out.

Did setup successfully complete on the IDH node?
